sec_certs/dataset.py

```diff
@@ -884,44 +892,50 @@ def convert_all_pdfs(self):
         ]
         cert_processing.process_parallel(FIPSCertificate.convert_pdf_file, tuples, constants.N_THREADS)
 
-    def get_certs_from_web(self, redo: bool = False, json_file: Optional[Path] = None):
+    def prepare_dataset(self, test: Optional[Path] = None):
```

Comment:
All public methods in `CCDataset` have an optional argument `update_json: bool = True` that, just before the function ends, updates the dataset json, so that you don't have to invoke it manually as you do in `fips_oop_demo.py`. I suggest you follow the same approach for all public methods in your class.